### PR TITLE
add new action hooks for better intigration and log imported and updated products

### PR DESCRIPTION
**Pull request by mzrahmani**
in order to extend the functionality  of the importer( like make all products that are not imported neither updated 0 stock ) and be able to integrated with multi-vendor solution, the following was added:
- log imported and updated products
- new action hooks was added

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
in order to extend the functionality  of the importer( like make all products that are not imported neither updated 0 stock ) and be able to integrated with multi-vendor solution, the following was added:
- log imported and updated products
- new action hooks was added
Closes # .

### How to test the changes in this Pull Request:

1. test the action hooks

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Create user_meta 'product_import_log' to log imported and updated products
add actions hooks:
- woocommerce_product_importer_batch_done
- woocommerce_product_importer_done

### CODE REVIEW

File: includes/admin/class-wc-admin-importers.php [hunk 1]
```diff
@@ -227,16 +227,21 @@ public function do_ajax_product_import() {
...
 
 		$importer         = WC_Product_CSV_Importer_Controller::get_importer( $file, $params );
 		$results          = $importer->import();
 		$percent_complete = $importer->get_percent_complete();
 		$error_log        = array_merge( $error_log, $results['failed'], $results['skipped'] );
+		$import_log['imported']        = array_merge( $import_log['imported'], $results['imported'] );
+		$import_log['updated']        = array_merge( $import_log['updated'], $results['updated']);
 
 		update_user_option( get_current_user_id(), 'product_import_error_log', $error_log );
+		update_user_option( get_current_user_id(), 'product_import_log', $import_log );
```
Comment (claudiosanches) [inline includes/admin/class-wc-admin-importers.php:244]: It get update, but there's no real use on WooCommerce for it.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Hello,

The current change may help your use case, but there's no use on WooCommerce.
Maybe better include more hooks or filters to allow you customize like you want, but in the format this Pull Request is today will not be acceptable.

**Comment by rodrigoprimo** [comment]
Closing due to lack of feedback. If this is still something that you would like to work on, please leave a comment and we will reopen the PR.

File: includes/admin/class-wc-admin-importers.php (hunk 1 above)
Comment (mzrahmani) [reply includes/admin/class-wc-admin-importers.php:244]: I can see the use for this hooks could be in many ways like:
- importing product details from third party source  
- import media from other online resource 
- apply dynamic changes during the import export 

so as the importing process is going, you have the import log to apply the dynamic changes or make API Call for further data during the Ajax call.

main use would be for plugins like multi vendor , multi language or auto listing solutions. usesare not limited for what i mentioned.

extending default woocommerce importer is important to provide integration between multiple advanced plugins which now each had created their own solutions.

**Comment by rodrigoprimo** [comment]
Reopening after the author provided more information.

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Thanks for your contribution, but this one seems to exclusive to your implementation, those options aren't used at all on WooCommerce in the format you are inserting, and could be better to introduce new hooks and filters to achieve it.
I'm closing for now since we shouldn't be accepting it on this format for now.
Thanks.

**Pull request closed without merging**